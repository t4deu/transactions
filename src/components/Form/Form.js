// @flow
import React, { Component } from 'react';
import { View } from 'react-native';
import { Toolbar, Button, Subheader } from 'react-native-material-ui';
import { TextField } from 'react-native-material-textfield';

import { today } from '../../Utils';

import styles from './Form.styles';

type Props = {
  title: string,
  onSubmit: Function,
  onBack: Function,
};

type State = {
  amount: string,
  description: string,
};

export default class Form extends Component<Props, State> {
  state = {
    amount: '',
    description: '',
  };

  handleAmountChange = (amount: string) => {
    this.setState({ amount });
  };

  handleSubmit = () => {
    const { amount, description } = this.state;

    this.props.onSubmit({
      amount: parseFloat(amount),
      description,
    });
  };

  goBack = () => {
    this.props.onBack();
  };

  render() {
    return (
      <View style={styles.container}>
        <Toolbar
          centerElement={this.props.title}
          leftElement="arrow-back"
          onLeftElementPress={this.goBack()}
          rightElement={<Button text="Adicionar" style={{ text: { color: 'white' } }} />}
        />
        <Subheader text={today()} />

        <View style={styles.form}>
          <TextField
            label="Valor"
            value={this.state.amount}
            onChangeText={this.handleAmountChange}
          />

          <TextField
            label="Descrição"
            value={`${this.state.description}`}
            onChangeText={amount => this.setState({ amount })}
          />
        </View>

        <Button raised primary icon="done" text="Adicionar" />
      </View>
    );
  }
}
