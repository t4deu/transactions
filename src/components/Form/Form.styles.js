export default {
  container: {
    flex: 1,
    padding: 20,
  },
  form: {
    flex: 1,
  },
};
