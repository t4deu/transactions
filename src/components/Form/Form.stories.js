import React from 'react';
import { storiesOf } from '@storybook/react-native';

import Form from './Form';

storiesOf('Form', module)
  .add('Add Credit', () => (
    <Form title="CrÃ©dito" onBack={() => {}} onSubmit={() => {}} />
  ));
