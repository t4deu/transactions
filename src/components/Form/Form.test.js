import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Form from './Form';

describe('<Form>', () => {
  it('renders without issues', () => {
    const component = shallow(<Form title="CrÃ©dito" onBack={() => {}} onSubmit={() => {}} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
