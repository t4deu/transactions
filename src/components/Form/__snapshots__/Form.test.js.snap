// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form> renders without issues 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Toolbar
    centerElement="Crédito"
    hidden={false}
    isSearchActive={false}
    leftElement="arrow-back"
    onLeftElementPress={null}
    onPress={null}
    onRightElementPress={null}
    rightElement={
      <Button
        accent={false}
        disabled={false}
        icon={null}
        onLongPress={null}
        onPress={null}
        primary={false}
        raised={false}
        style={
          Object {
            "text": Object {
              "color": "white",
            },
          }
        }
        text="Adicionar"
        upperCase={true}
      />
    }
    searchable={null}
    size={24}
    style={Object {}}
  />
  <Subheader
    inset={false}
    lines={1}
    style={Object {}}
    text="Quarta-feira, 28 de março"
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextField
      activeLineWidth={2}
      animationDuration={225}
      autoCapitalize="sentences"
      baseColor="rgba(0, 0, 0, .38)"
      disableFullscreenUI={true}
      disabled={false}
      disabledLineType="dotted"
      disabledLineWidth={1}
      editable={true}
      errorColor="rgb(213, 0, 0)"
      fontSize={16}
      inputContainerPadding={8}
      label="Valor"
      labelFontSize={12}
      labelHeight={32}
      labelPadding={4}
      lineWidth={0.5}
      onChangeText={[Function]}
      textColor="rgba(0, 0, 0, .87)"
      tintColor="rgb(0, 145, 234)"
      titleFontSize={12}
      underlineColorAndroid="transparent"
      value=""
    />
    <TextField
      activeLineWidth={2}
      animationDuration={225}
      autoCapitalize="sentences"
      baseColor="rgba(0, 0, 0, .38)"
      disableFullscreenUI={true}
      disabled={false}
      disabledLineType="dotted"
      disabledLineWidth={1}
      editable={true}
      errorColor="rgb(213, 0, 0)"
      fontSize={16}
      inputContainerPadding={8}
      label="Descrição"
      labelFontSize={12}
      labelHeight={32}
      labelPadding={4}
      lineWidth={0.5}
      onChangeText={[Function]}
      textColor="rgba(0, 0, 0, .87)"
      tintColor="rgb(0, 145, 234)"
      titleFontSize={12}
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <Button
    accent={false}
    disabled={false}
    icon="done"
    onLongPress={null}
    onPress={null}
    primary={true}
    raised={true}
    style={Object {}}
    text="Adicionar"
    upperCase={true}
  />
</View>
`;
